extra/hip_wrapper.py

```diff
@@ -200,21 +200,23 @@ def getStructTypeForArgs(*args):
         fields.append((f'field{idx}', ctypes.c_void_p))
     return __get_struct(types, fields)
 
-  def updateKernelNodeParams(npwrapper:kernelNodeParamsWrapper, *args, grid=(1,1,1), block=(1,1,1), updated_args=None):
-    _, struct, _ = npwrapper.context
-    if updated_args is not None:
-      for i in updated_args:
-        setattr(struct, f'field{i}', (args[i] if args[i].__class__ is int else args[i]._buf))
-    else:
-      for i,d in enumerate(args):
-        setattr(struct, f'field{i}', (d if d.__class__ is int else d._buf))
+  def setKernelNodeParam(npwrapper:kernelNodeParamsWrapper, arg, id):
```

Comment:
Actually should these helpers be here at all? Pushing complexity to extra is bad...

Reply:
I have removed everything tinygrad-specific from the wrapper (such as ._buf). The wrapper still contains helper functions, but I believe these are more closely related to HIP. Should I pull out more?